src/oop/user_defined_classes/hard/q4/MainTest.java

```diff
@@ -0,0 +1,98 @@
+package oop.user_defined_classes.hard.q4;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String testClassName = "Laundromat";
+    private final String intAttributeName1 = "gymClothes";
+    private final String methodName = "doLaundry";
+    private ObjectTest classInstance;
+
+    @BeforeEach
+    public void setUp() {
+        String testClassString = "oop.user_defined_classes.hard.q4." + testClassName;
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void laundromatClassHasCorrectAttributes() {
+        String missingAttributeMessage =
+                "The %s attribute could not be found in your %s class. Please make sure you have added it, it is spelled correctly, and is of the correct type";
+        String incorrectModifierMessage = "Your %s attribute in the %s class does not have the correct visibility modifier.";
+        assertTrue(classInstance.hasField(intAttributeName1, int.class),
+                String.format(missingAttributeMessage, intAttributeName1, testClassName));
+        assertTrue(classInstance.hasModifier(intAttributeName1, "private"),
+                String.format(incorrectModifierMessage, intAttributeName1, testClassName));
+    }
+
+    @Test
+    public void laundromatClassHasRequiredConstructor() {
+        Class<?>[] arguments = {
+                int.class
+        };
+        assertTrue(classInstance.hasConstructor(arguments),
+                String.format("Your %s class is missing a required constructor.", testClassName));
```

Comment:
Maybe add "Please check that you have added the constructor and it has the correct parameters."